classes/autoptimizeExtra.php

```diff
@@ -266,6 +266,12 @@ public function filter_optimize_google_fonts( $in )
         if ( '2' === $options['autoptimize_extra_radio_field_4'] ) {
             // Remove Google Fonts.
             unset( $fonts_collection );
+            
+            // Real Cookie Banner compatibility: Do not show this markup as Google Font result in the scanner
+            if (isset($_GET['rcb-scan'])) {
+                return str_replace('<link', '<link consent-scanner-skip', $in, 1);
```

Comment:
Hey @futtta !

Sorry, I created the PR too quickly. Of course it won't work like that, because the `str_replace` has to happen on the match of `preg_match_all`, it would be quasi a refactor to `preg_replace_callback` necessary:

https://github.com/futtta/autoptimize/blob/536be4d19d4019b5bbd01dd0065a6b5a28cbb88c/classes/autoptimizeExtra.php#L241

What do you think about my first solution proposal?

> Use `autoptimize_filter_html_before_minify` instead of `autoptimize_html_after_minify` for the `filter_optimize_google_fonts` function

Reply:
Although it might work, I'm _very_ reluctant to make such potentially profound changes to how AO works I'm afraid @matzeeable . 

An alternative solution; why don't you check the AO Extra options (`$options = get_option( 'autoptimize_extra_settings' );`) and `if '2' === $options['autoptimize_extra_radio_field_4']` then Google Fonts are to be ignored even if found pre-AO-optimization?

Reply:
Hey @futtta !

> An alternative solution; why don't you check the AO Extra options [...]

From our experience, I can say that there are countless ways to include resources in WordPress, even if they do not conform to the standard. I don't think it makes sense to rely on this option.

Do you have any idea how we could mark (instead of remove) the removable Google Fonts import? In general, we need to check for the `$_GET['rcb-scan']` parameter, as this indicates that the RCB scanner is running:

https://github.com/futtta/autoptimize/blob/536be4d19d4019b5bbd01dd0065a6b5a28cbb88c/classes/autoptimizeExtra.php#L274-L275

Reply:
sorry for the late reply, but I could add a filter in https://github.com/futtta/autoptimize/blob/beta/classes/autoptimizeExtra.php#L177 to allow one to disable Gfont removal, hooking that filter into code that check the querystring for rcb-scan is trivial, but marking the fonts at that point would have to be done on your side?

an alternative might be to on your end instantiate `autoptimizeExtra` and hook `filter_optimize_google_fonts` into `autoptimize_filter_html_before_minify` when running your scanner ?